indexing-service/src/main/java/org/apache/druid/indexing/common/actions/CheckPointDataSourceMetadataAction.java

```diff
@@ -75,6 +75,28 @@ public SeekableStreamDataSourceMetadata getPreviousCheckPoint()
     return checkpointMetadata;
   }
 
+  // For backwards compatibility
```

Comment:
Can add a unit test for the serialization/deserialization between old and new DTOs

Reply:
yeah, was trying to add a serde test for `CheckPointDataSourceMetadataAction`, but wasn't easy to construct an object because of abstract `SeekableStreamDataSourceMetadata` in this package.

Reply:
One option is to manually create a JSON string and then deserialize it to a different version. If you don't want to manually create the JSON, then one option is to add a class in the test file that is the old DTO.

Reply:
Added a serde test for kafka